### DISCUSSION TITLE
Porting Linux drivers

### DISCUSSION THREAD

**Discussion by jefffhaynes** [zephyrproject-rtos/zephyr#53170] [Q&A]
I guess this is a multipart question.  I'm interested in using a temperature sensor that isn't currently supported by Zephyr (probably the SA56004).  I've noticed that there is a linux driver that supports a bunch of different temp sensors (source [here](https://elixir.bootlin.com/linux/latest/source/drivers/hwmon/lm90.c)) but I can't find any reference to any of the sensors within Zephyr or on here.  I'm a little surprised that no one has ever needed to use any of these parts before so my questions are:

- Am I missing something?
- Is porting linux drivers like this whole-sale to Zephyr something that's often done?
- Should I take a stab at porting the driver or is that a fool's errand and I should focus instead on just writing a driver for the device I need?

**Comment by towen** [comment]
I'd guess the main issue is licensing.. Linux is GPL, whereas Zephyr uses the more-permissive Apache-2.0 license. (i.e. closed-source commercial use of Zephyr is allowed, whereas any modifications to the Linux kernel must be re-published as GPL in order to be used commercially, at least in theory..)
So you are free to copy code wholesale from Linux for your own use and for others' use, but it wouldn't be accepted into the main Zephyr tree. You'd have to publish it as an out-of-tree, GPL-licensed driver. I don't know if there is any "common repository" for those.

**Comment by henrikbrixandersen** [comment]
Also, you may want to take a look at the existing lm75 and lm77 Zephyr drivers:
- https://github.com/zephyrproject-rtos/zephyr/tree/main/drivers/sensor/lm75
- https://github.com/zephyrproject-rtos/zephyr/tree/main/drivers/sensor/lm77
